### PR TITLE
PC-1017: Add PR template

### PR DESCRIPTION
**Pull request by samyou-softwire**
# Link to Ticket

[Link to Jira ticket](https://beisdigital.atlassian.net/browse/PC-1017)

# Description

this adds a pull request template, with additional check boxes for extracting translations & website flow

since this is not yet on main, I copied the template in manually to check it all looks fine

# Checklist

- [x] I have made any necessary updates to the documentation
- [x] I have checked there are no unnecessary IDE warnings in this PR
- [x] I have checked there are no unintentional line ending changes
- [x] I have added tests where applicable
- [x] If I have made any changes to the code, I have used the IDE auto-formatter on it
- [x] If I have made any changes to Python files (even if not changing any content strings), I have run `make extract-translations`
- [x] If I have made any changes to website flow, I have updated the [Flow Miro Board](https://miro.com/app/board/uXjVNyomz0g=/)
- [x] If I have made any changes to website flow, I have checked forward and back behaviour is still consistent

### CODE REVIEW

**Review by jdgage — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+ <!-- Add a brief description of the change(s) you have made --> 
+
+# Checklist
+
+- [ ] I have made any necessary updates to the documentation
+- [ ] I have checked there are no unnecessary IDE warnings in this PR
+- [ ] I have checked there are no unintentional line ending changes
+- [ ] I have added tests where applicable
+- [ ] If I have made any changes to python files, I have run `make extract-translations`.
```
Comment (jdgage) [inline .github/pull_request_template.md]: Is this all Python files, or can we be more specific?

Comment (samyou-softwire) [reply .github/pull_request_template.md]: essentially. in theory a more correct description is 'all Python files that contain a string to be translated, even if the string wasn't edited' but in practice this is most python files so I left as all, no harm as regenerating translations doesn't lead to any changes

Comment (jdgage) [reply .github/pull_request_template.md]: Fair enough. In that case maybe it's worth emphasising the point:
```
- [ ] If I have made any changes to Python files (even if not changing any content strings), I have run `make extract-translations`.
```
?

Comment (samyou-softwire) [reply .github/pull_request_template.md]: sure

**Review by jdgage — APPROVED** [review]

**Pull request merged**